### ISSUE TITLE
Virus Total does not like µTox

### ISSUE THREAD

**Issue by ikabod** [uTox/uTox#1373]
### Overview

1. uTox version: Do not know - just downloaded it but did not installed it
2. Operating system: Windows 64

### Issue

Write your issue here.

On Virus Total 6 engines finds viruses etc, see https://www.virustotal.com/gui/file/96c9e2d0c595bf0a667ad361302cd9fce453b803cb0c42ba561f3b8ed626b240/detection

May be this could be bypasses by creating a zip version? This would have the advantage that this zip version could be preconfigured with my own tox id and given to users with less computer experience and/or to launch a quick support session. I could provide this zip file on my web page, user download it, extract it and double click the contained exe file. Than he is ready to connect to me.

**Comment by avoidr** [comment]
I don't understand your issue. What does the title have to do with what you want to do?

**Comment by ikabod** [comment]
Now you have the complete program code in one single exe file. I assume on the first run it will extract files to the folder where the exe is located in and/or create a profile in the users folder etc.

This behavior and/or the behavior of the single exe file at all is recognized by some Antivirus software as "evil".

If you put most of the code into one (or more) dll file the exe could be very small and have no longer to do the tasks above since the folder structure already exists in a zip file and all needed files also. To be portable the dll file has to be of a type which does not need registering (regsvr32).

It than would also be possible to put tox profile files into this zip package and/or adjust an ini-file to preconfigure the program.

In short: Use the new, small exe file to call functions from the dll file.

I guess this new exe file would no longer be detected as "evil".

**Comment by GrayHatter** [comment]
> Now you have the complete program code in one single exe file. I assume on the first run it will extract files to the folder where the exe is located in and/or create a profile in the users folder etc.

No, that's one of the features of uTox, it runs from just a single exe.

> This behavior and/or the behavior of the single exe file at all is recognized by some Antivirus software as "evil".

I hope not, that's a really bad heuristic. But Toxcore is used by some viruses, so I'm assuming because uTox also uses Toxcore, that lazy antivirus software just assumes everything with tox is a virus.

> If you put most of the code into one (or more) dll file the exe could be very small and have no longer to do the tasks above since the folder structure already exists in a zip file and all needed files also. To be portable the dll file has to be of a type which does not need registering (regsvr32).

That would be very annoying to use...

> It than would also be possible to put tox profile files into this zip package and/or adjust an ini-file to preconfigure the program.

That's actually an interesting idea, you could build uTox with a profile built in...

> In short: Use the new, small exe file to call functions from the dll file.
> I guess this new exe file would no longer be detected as "evil".

 Nah, there will always be bad AV who's authors aren't very good...

**Comment by ikabod** [comment]
Hi again,

even if there might be false positives with the current exe file, it will prevent spreading your tool widely. You should do something about that. The qTox-Installer and the qtox.exe are ok with Virus Total.

The way with the dll and the zip file could be handled by the Portable Apps File Format for better usability (see https://portableapps.com/about).

A lot of Windows user (me included) works almost only with portable apps, running from a micro sd-card. That is really a great advantage compared to Linux. Take the sd-card to any Windows computer and you can nearly work as your own.

I would like to recommend your client on my webpage(s) because it is very lightwight instead of qTox (20 MB zipped, 40 MB unzipped). But I can not as long as the anti virus software shows that it is some kind of Virus/Trojaner, even if it is not.

I hope you will consider my point of view. Thanks.

**Comment by GrayHatter** [comment]
@ikabod I see, and agree with your point of view. 

But I'm not willing to put in the time to talk to the small AV companies to try to correct them. If there was some technical solution, I'd be more than happy to implement it. But some small/lazy AV company isn't a uTox bug, perhaps you should open a issue with the AV companies marking uTox as a false positive.
